code/scripting/api/libs/hud.cpp

```diff
@@ -61,15 +79,15 @@ ADE_FUNC(setHUDGaugeColor, l_HUD,
 	int r = 0;
 	int g = 0;
 	int b = 0;
-	int a = 0;
+	int a = 255;
 
 	if(!ade_get_args(L, "i|iiii", &idx, &r, &g, &b, &a))
 		return ADE_RETURN_FALSE;
 
-	if ((idx < 0) || (idx >= NUM_HUD_GAUGES))
+	if ((idx < 0) || (idx >= default_hud_gauges.size()))
 		return ADE_RETURN_FALSE;
 
-	gr_init_alphacolor(&HUD_config.clr[idx], r, g, b, a);
```

Comment:
These changes (in both functions) do not cover the use case of saving the updated values to the pilot file which is currently done for the `HUD_config` object.

To support both use cases you could introduce new functions with your changed code to support this different use case.

Reply:
Moved the new functionality to a separate function. Will still modify the old documentation as it is currently misleading